tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_test.ts

```diff
@@ -254,4 +264,74 @@ describe('regex_edit_dialog', () => {
     const input = fixture.debugElement.query(By.css('input'));
     expect(input.nativeElement.value).toBe('(train|eval)');
   });
+
+  describe('live grouping result preview', () => {
```

Comment:
I actually do not trust the code I wrote. Would you mind adding a small spec that tests whether the input is populated with regex string from the store initially? If you could add another spec that tests for its changes would be nice: like if store dispatches change and if the regex string in the store has changed, do we want to update the input? Probably not.

Reply:
> whether the input is populated with regex string from the store initially
added (not yet push)
> if the regex string in the store has changed
a couple of questions:
- why this might happen?
- at this timing (regex string in the store had changed) do we already have tentativeString?
`merge` takes the last value so if the store changed value after tentativeString changes, `groupByRegexString$` would be the store value (`this.store.select(getColorGroupRegexString)` then)?
my own interpretation of `groupByRegexString$`: get the value from store as the initial value, then always be `tentativeString$` afterward.

Reply:
> why this might happen?

it does not happen but as a view component, you do not want to make too much assumptions but, instead, just follow the contract between a store and the view (i.e., store can emit change for whatever reason it wants). 

> my own interpretation of groupByRegexString$: get the value from store as the initial value, then always be tentativeString$ afterward.

Yes, that was my intention when I wrote that code. However, I don't want to trust brain so I would want a test that test exactly that assumption :)